airflow/contrib/hooks/databricks_hook.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import requests
+
+from airflow import __version__
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+from requests import exceptions as requests_exceptions
+
+
+try:
+    from urllib import parse as urlparse
+except ImportError:
+    import urlparse
+
+
+SUBMIT_RUN_ENDPOINT = ('POST', 'api/2.0/jobs/runs/submit')
+GET_RUN_ENDPOINT = ('GET', 'api/2.0/jobs/runs/get')
+USER_AGENT_HEADER = {'user-agent': 'airflow-{v}'.format(v=__version__)}
+
+
+class DatabricksHook(BaseHook):
+    """
+    Interact with Databricks.
+    """
+    def __init__(
+            self,
+            databricks_conn_id='databricks_default',
+            timeout_seconds=180,
+            retry_limit=3):
```

Comment:
Airflow already has the concept of retries at the Operator level. While temporary outages are possible, ideally the user would decide on their own how to handle this, what if the workload is time sensitive and they would rather skip this run and move on to the next, rather than wait for retries?

I am not opposed to sensible defaults in the hook, but ideally as a user I should be able to override those via config/params in my workflow.